src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp

```diff
@@ -45,6 +45,14 @@ JSONEachRowRowInputFormat::JSONEachRowRowInputFormat(
 {
     const auto & header = getPort().getHeader();
     name_map = header.getNamesToIndexesMap();
+    if (format_settings_.json.ignore_key_case)
+    {
+        for (auto it = name_map.begin(); it != name_map.end(); ++it)
+        {
+            StringRef key = it->first;
+            transformFieldNameToLowerCase(key);
```

Comment:
As it seems to me, you try to mutate key of a hash table while iterating on it. Usually mutating of a key breaks insides of an iterator of a data structure